src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+"""Transform raw AEO tables into normalized assets.
+
+Raw AEO tables often contain many different types of data which are split out
+along different dimensions. For example, one table may contain generation
+split out by fuel type as well as prices split out by service category.
+
+As a result, we need to split these large tables into smaller tables that have
+more uniform data, which we do by filtering the large table to its relevant
+subsets, and then transforming some human-readable string fields into useful
+metadata fields.
+"""
+
+import numpy as np
+import pandas as pd
+from dagster import asset
+
+from pudl.helpers import apply_pudl_dtypes
+
+
+def __sanitize_string(series: pd.Series) -> pd.Series:
+    return series.str.lower().str.strip().str.replace(r"\W+", "_", regex=True)
+
+
+def get_series_info(series_name: pd.Series) -> pd.DataFrame:
```

Comment:
what is a `series_name`? is it like a weird series of a kind of index/pk of whatever table? ah i see this what they have named this column.

Reply:
Yeah, we're combining a bunch of "series" and the series name contains a bunch of useful metadata that they "could" have just put in separate JSON fields but 🤷